python/ray/data/tests/test_object_gc.py

```diff
@@ -234,6 +233,22 @@ def consume(p):
     meminfo = memory_summary(ctx.address_info["address"], stats_only=True)
     assert "Spilled" not in meminfo, meminfo
 
+def test_bytes_spilled():
+    # The object store is about 90MB.
+    ray.init(object_store_memory=90e6)
+    # The size of dataset is 500*(80*80*4)*8B, about 100MB.
+    ds = ray.data.range_tensor(500, shape=(80, 80, 4), parallelism=100).materialize()
+    
+    assert ds._get_stats_summary().bytes_spilled > 0
```

Comment:
nit: can we make use of `check_no_spill(...)`/ `not check_no_spill(...)` to implement the spill check, similar to other tests in this file?

Reply:
Yeah I'll add `check_no_spill` to these tests, but that function doesn't cover the new code path introduced here. This tests retrieving the stats to `DatasetStats` after plan execution. Maybe, this test should be somewhere else?